### DISCUSSION TITLE
Sparring matches

### DISCUSSION THREAD

**Discussion by lucasmeneghin** [SebLague/Chess-Challenge#77] [Ideas]
Can someone think of a relatively easy way we can do sparring matches without having to share our code? So we can test our bots before submitting

**Comment by HuskiGames** [comment]
Both teams can just play their bot as bot vs human. Then the human plays the moves of the other bot. This can be done over a discord call and won't require any sharing of code.

**Reply by lucasmeneghin** [reply]
Thats a good idea tho it would impact the time
Maybe for a longer time control, but for bullet this could get tricky
A somewhat more complex solution would be to make either bot an API and make evilbot call that api on the other team's client
This would also favor the hosts time, but i guess much less
This shouldnt be too hard to implement aswell, if i keep developing my bot i might do it

**Reply by manfromfms** [reply]
I think that there would be no impact on time. Each bot has its own time limit and when you are delivering move from one bot to another in any way (discord or socket) they all are stopped.

**Reply by lucasmeneghin** [reply]
What do you mean, if my bot makes a move and it takes 1 second to calculate, then after its finished calculating it will start using the opponents time on my pc, while on yours it will still be using my time. so after i tell you my move and you input it, a ~2s delay lets say, my bot is now with 59 seconds on my pc, yours with 58s, and on your pc my bot has 57 seconds and yours has 1 min and will start thinking now

**Reply by manfromfms** [reply]
I mean that if you play via discord both people chose to play "Human vs MyBot". Human gets infinite time there so after one bot makes a move you can even fly by plane to other pc and make another move. All this time while you are traveling both bots timers will be paused.

**Reply by lucasmeneghin** [reply]
You're correct, i'm sorry

**Comment by draeg82** [comment]


**Reply by lucasmeneghin** [reply]
"without having to share our code"
